DataStructures/Heaps/BinomialMinHeap.cs

```diff
@@ -182,13 +192,14 @@ public bool IsEmpty
             get { return (_size == 0); }
         }
 
+
         /// <summary>
         /// Initializes this heap with a collection of elements.
         /// </summary>
         public void Initialize(IList<T> newCollection)
         {
             if (newCollection == null)
-                throw new ArgumentNullException();
+                throw new ArgumentNullException($"The provided parameter was null");
```

Comment:
Why are we using parameterizable strings when there are no parameters?
This question also applies to the other exception messages without parameters.

Reply:
SonarLint suggested to extend the exception descreption so we made it open for future clarifications by added variables.

Reply:
I agree with @MarcellHarmaci, the "$" sign is unnecessary here.

The added message could be useful, but then the parameter name should be included.